dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -409,7 +422,14 @@ private void handlePotentialFormatEntry(Set<FileFormat> fileFormats, Object v) {
                     final Optional<SourceFile> sourceFileOptional = secondarySourceFiles.stream()
                             .filter(sf -> sf.getPath().equals(finalSecondaryFile)).findFirst();
                     final String content = sourceFileOptional.map(SourceFile::getContent).orElse(null);
-                    stepDockerRequirement = parseSecondaryFile(stepDockerRequirement, content, gson, yaml);
+                    try {
+                        Yaml safeSecondaryYaml = new Yaml(new SafeConstructor());
+                        // This should throw an exception if there are unexpected blocks
+                        safeSecondaryYaml.load(finalSecondaryFile);
+                        stepDockerRequirement = parseSecondaryFile(stepDockerRequirement, content, gson, yaml);
+                    } catch (Exception e) {
+                        LOG.info("Secondary descriptor uses unsafe types");
```

Comment:
1. Log the actual exception (second parameter).
2. Filename/step would be nice as well.
3. I think you should throw the exception. There's a catch at the bottom that this would ideal hit. I don't know if you can wrap it one of those exception classes, or if it's already one of them? But in any case, if there were a "regular" YAML error, we stop processing and throw a CustomWebApplication, so we should do the same here if it's a security error.